### PR TITLE
[inductor] Scalar accumulators for all reductions in large inner loops

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.17.0) (oldest at bottom):
* __->__ #196371

#190692 gave online softmax per-row scalar accumulators in large inner reduction loops. This extends them to every reduction of such a kernel when one or two online softmaxes are present, the cases that were swept: argmax/argmin and their value variants fold a per-block reduction into per-row state, and sum/prod/max/min/xor_sum reuse the strict-reduction loop structure with the reduction's combine function, casting each block's result back to the accumulator type because tl.sum widens narrow ints. Eligibility is all-or-nothing per kernel, since a kernel mixing scalar and vector accumulators spills on the large blocks the hint enables. Kernels without an online softmax are unchanged: lone arg reductions and plain sums have nothing to spill and measured slower on some shapes, so they wait for their own sweep. Under strict_signed_zero the whole kernel keeps the vector path. The flag and hint become triton.scalar_accumulators and AutotuneHint.SCALAR_ACCUMULATORS.

Review order: use_scalar_accumulators in triton.py, the reduction() branches, then triton_helpers.py.

Arg reductions also get max_with_first_index / min_with_first_index: max2/min2 followed by the smallest index attaining the result, instead of a tuple reduce with a NaN-aware combine. They apply only where the index alone is consumed and a whole tile is reduced at once (persistent kernels and the scalar per-block path), CUDA only and behind the same flag; max(dim)/min(dim) with a live value keep the tuple reduce, which returns the winning lane's signed zero.

GB300 bf16, measurement scripts not in tree: Domino cross entropy (#190718) 3.23 -> 1.71 ms at 4096x248320 and 3.43 -> 1.77 ms at 8192x128256 (hand-written Triton kernel: 2.03 / 2.07 ms); cross entropy fused with argmax 0.55x; online-softmax-only kernels unchanged; better-benchmark's persistent arg-reduction repros 0.88x geomean. The scalar configs autotune within a few percent of each other at small row counts, so fp32 sums in a scalar kernel can round differently between compiles; deterministic mode picks a fixed one.

Test Plan:

```bash
export TORCHINDUCTOR_FORCE_DISABLE_CACHES=1
python test/inductor/test_online_softmax.py
python test/inductor/test_triton_heuristics.py
python test/inductor/test_codegen_triton.py -k autotune_hints
python test/inductor/test_torchinductor.py -k "argm or reduction or max_min"
python test/inductor/test_torchinductor_strided_blocks.py -k 2d_reductions
lintrunner -a
```

All pass on a GB300 (CUDA 13.0, triton 3.8). Differential fuzzers over sum, amax, amin, argmax, argmin, max/min with indices, prod and fused combinations across four dtypes with NaN, inf and tied inputs matched the vector path and eager.

Authored with assistance from Claude.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -5893,12 +5947,17 @@ def _mask_value(value, default) -> CSEVariable:
                     self.body.writeline(
                         f"{accumulator} = tl.full({dense_size_str}, {default}, {acc_type})"
                     )
-                elif strict_reduction_loop:
+                elif strict_reduction_loop or scalar_loop:
```
Comment (karthickai) [inline torch/_inductor/codegen/triton.py:5950]: this breaks sum(dtype=torch.uint8) when fused with online softmax. the accumulator starts as tl.uint8, but the per-block tl.sum returns tl.uint32, so triton rejects the loop-carried type change

**Review by karthickai — APPROVED** [review]
LGTM, pls address the blocker before landing
